packages/dashboard/src/CdcDashboardComponent.tsx

```diff
@@ -265,6 +265,10 @@ export default function CdcDashboard({ initialState, isEditor = false, isDebug =
     loadAPIFilters(config.dashboard.sharedFilters)
   }, [isEditor, isPreview])
 
+  useEffect(() => {
+    updateDataFilters()
```

Comment:
I think if you move this into the first useEffect right after loadAPIFilters it would also work. Did you try that?

Reply:
I think this is related to this issue you mentioned on 3. in dev-8409

Reply:
Yes, I had tried that, but the useEffect only has isEditor and isPreview as dependencies, so it doesn't fire when the dashboard multi-tab is changed. I can combine the useEffects possibly, including state.config?.activeDashboard on the first useEffect, but then API filters will reload every time the multi-tab changes.

Reply:
> I think this is related to this issue you mentioned on 3. in dev-8409

I don't believe so, 3 in 8409 is related to the value not being checked for null in packages/dashboard/src/helpers/generateValuesForFilter.ts

Reply:
> I think if you move this into the first useEffect right after loadAPIFilters it would also work. Did you try that?

Yes, I had tried that, but the useEffect only has isEditor and isPreview as dependencies, so it doesn't fire when the dashboard multi-tab is changed. I can combine the useEffects possibly, including state.config?.activeDashboard on the first useEffect, but then API filters will reload every time the multi-tab changes.

Reply:
@joshlacey I combined the useeffects in the latest commit, let me know if looks better now

Reply:
you should update the shared filters and then pass the result to loadAPIFilters. that's what I'm doing in the dashboard filters refactor branch